### PR TITLE
Add Python based Allied Vision code

### PR DESCRIPTION
**Pull request by neildhar**
Adapt pymba python library to do continuous imaging and publishing with ROS. See README for running instructions

### CODE REVIEW

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [inline src/avt_camera/scripts/mono_camera.py]: Should check if cameras were found

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [inline src/avt_camera/scripts/mono_camera.py]: Somehow check if the same camera is requested and give a reasonable error message if that happened

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [inline src/avt_camera/scripts/mono_camera.py]: Error, exit with non zero exit code

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [inline src/avt_camera/scripts/mono_camera.py]: Error, exit with non zero exit code

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [inline src/avt_camera/scripts/mono_camera.py]: Where is documentation for these parameters?

File: src/avt_camera/scripts/test.py
Comment (davidAmiron) [inline src/avt_camera/scripts/test.py]: Remove this file

**Review by davidAmiron — DISMISSED** [review]
Remove test.py

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [reply src/avt_camera/scripts/mono_camera.py]: Should already do that

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [reply src/avt_camera/scripts/mono_camera.py]: never mind

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [reply src/avt_camera/scripts/mono_camera.py]: Look at vimba viewer, ctrl-f in pymba

File: src/avt_camera/scripts/mono_camera.py
Comment (davidAmiron) [reply src/avt_camera/scripts/mono_camera.py]: neil: "try catch on opencamera"

File: src/avt_camera/scripts/stereo_camera.py
Comment (neildhar) [inline src/avt_camera/scripts/stereo_camera.py]: there's a lot of code duplication here, consider using a function or a loop to do the setup on both cameras simultaneously

**Comment by davidAmiron** [comment]
This is temporarily in backlog, we are getting the jetson working with GPUs until we can test this some more.

File: src/avt_camera/scripts/stereo_camera.py
Comment (muthuArivoli) [reply src/avt_camera/scripts/stereo_camera.py]: Most, if not all duplication was removed in 3bec872. Last bit (between monocamera and synchronouscamera) can be figured out later, if we decide that synchronous camera is good enough without mono camera.

**Comment by muthuArivoli** [comment]
Things to do before merge
- [ ] Test single camera, especially whether output has been formatted appropriately for CV
- [ ] Test synchronized camera
- [ ] Test throwing exceptions 
- [ ] Add calibration for downward camera
- [ ] Add launch file for downward camera

Notes:
This new code will need to be tested thoroughly. It should be functionally equivalent to the code from the last time we tested except for two things:

1. An exception has been removed in the vimba_frame header file, specifically the `waitFrameCapture` function. Previously, we had the pymba library in this package, and had manually added that exception in. Since the library has been moved to the docker image, we don't have access to the library files, so it currently will not throw an exception. If this causes problems, then we can process the exception in our code itself, by changing `self._frame.waitFrameCapture()` to 
```
err = self._frame.waitFrameCapture()
if err != 0:
    raise VimbaException(err)
```

2. The code should now be able to detect if a camera is requested a second time, and will exit with error. Not completely sure if this will work, will definitely need to test for false positives on restarting a camera. If there are indeed false positives, then the only other way I can currently see is to get the private variable  `_cameras` from our Vimba instance. Using this method, we would check if the camera_id is in the keys already, immediately before `self._c0 = vimba.getCamera(self._camera_id)`.  
```
if self._camera_id in vimba._cameras:
    sys.exit(1)
```
This is definitely not good coding style, accessing a private variable like this, but I was not able to find any other way.

File: onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml
Comment (davidAmiron) [inline onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml:1]: How were these generated? Do they work?

File: onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml
Comment (davidAmiron) [inline onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml:9]: They are yaml files, can we remove the `!!python/unicode` from all the lines?

File: onboard/catkin_ws/src/avt_camera/scripts/synchronous_camera.py
Comment (davidAmiron) [inline onboard/catkin_ws/src/avt_camera/scripts/synchronous_camera.py:9]: Rename to SynchronizedCameras

File: onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml
Comment (muthuArivoli) [reply onboard/catkin_ws/src/avt_camera/calibrations/DEV_000F315C1ED5.yaml]: They do work. I believe they were generated using the camera calibration package in ROS.

**Comment by muthuArivoli** [comment]
This has been tested with both the left and right camera individually, and with them together for stereo vision. All the exceptions have been tested like trying to call a camera twice and trying to run without a camera attached. This should be good to merge.

**Review by ShaanGondalia — APPROVED** [review]
LGTM

**Review by rabinowitz — CHANGES_REQUESTED** [review]
LGTM! One addition: please update the list of packages in the repo's main README to include this new one.

**Review by rabinowitz — APPROVED** [review]
LGTM! Squash and merge when ready.

**Review by muthuArivoli — APPROVED** [review]
This is finally landing today

**Pull request merged**
